microsoft/testsuites/dpdk/dpdksuite.py

```diff
@@ -85,6 +85,31 @@ def verify_dpdk_build_failsafe(
     ) -> None:
         self._verify_dpdk_build(node, log, variables, "failsafe")
 
+    @TestCaseMetadata(
+        description="""
+            test sriov failsafe during vf revoke (receive side)
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_nic_count=2,
+            network_interface=Sriov(),
+            min_count=2,
+        ),
+    )
+    def verify_dpdk_sriov_rescind_failover_receiver(
+        self, environment: Environment, log: Logger, variables: Dict[str, Any]
+    ) -> None:
+
+        test_kits = _init_nodes_concurrent(environment, log, variables, "failsafe")
+        sender, receiver = test_kits
+
+        kit_cmd_pairs = generate_send_receive_run_info("failsafe", sender, receiver)
+
+        _run_testpmd_concurrent(kit_cmd_pairs, 60 * 5, log, rescind_sriov=True)
```

Comment:
Does it mean this test case will run 300 seconds? Can it be quicker, like 30 sec even less?

Reply:
Nope, the device removal and re-attach can each take a full minute to hit the VM and get processed. Maybe we can cut it to 4 minutes instead of 5.

Reply:
Hey! we can cut it to 3 :)

Reply:
The hard code time either not stable, or take longer time in usual. It's better to check some signal, and start next step, once some condition is met. It means we can set some timeout here like 10 minutes, but in most cases, it will completed in 1 minute.

Reply:
I can do that, will ping you when it's ready